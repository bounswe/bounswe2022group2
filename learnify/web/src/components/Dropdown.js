import * as React from 'react';
import './component_styles.css';

export default function Dropdown(props) {
    const [isActivated, setIsActivated] = React.useState(false);
    const options = [    "Math and Logic",
    "Physical Sciences and Engineering",
    "Social Sciences",
    "Programming",
    "Visual Arts",
    "Music",
    "Gastronomy and Culinary",
    "Business",
    "Sports",
    "Performance Arts",
    "Information Technologies",
    "Personal Development",
    "Language Learning",
    "Home Economics",
    "Parenting",
    "Lifestyle"];
    const [selected, setSelected] = React.useState(null);
    return (
        <div className='dropdown'>
            <div>
            <button 
            className='dropdown-btn'
             onClick={(e) => setIsActivated(!isActivated)}>
                {selected ? selected : 'Select Category'}
            </button>
            </div>
            {isActivated && (
                <div className='dropdown-content'>
                    {options.map((option) => (
                        <div 
                            className='dropdown-item'
                            onClick={(e) => {
                                setSelected(option);
                                setIsActivated(false);
                            }}
                        >
                            {option}
                </div>
                    ))}
                </div>
            )}
            
        </div>
    );
  }