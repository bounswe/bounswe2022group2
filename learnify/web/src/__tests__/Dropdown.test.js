import React from "react";
import Dropdown from "./../components/Dropdown";
import { screen, render, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

describe("Dropdown", () => {
    test('should render component layout correctly', async () => {
        const component = renderer.create(<BrowserRouter><Dropdown /></BrowserRouter>).toJSON();
        expect(component).toMatchSnapshot();
    });
}   );

test('dropdown renders correctly', async () => {
    render(<BrowserRouter><Dropdown /></BrowserRouter>);
    const dropdown = screen.getByText(/Select Category/i);
    expect(dropdown).toBeInTheDocument();
});