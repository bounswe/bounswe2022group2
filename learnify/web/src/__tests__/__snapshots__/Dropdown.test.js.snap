// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render component layout correctly 1`] = `
<div
  className="dropdown"
>
  <div>
    <button
      className="dropdown-btn"
      onClick={[Function]}
    >
      Select Category
    </button>
  </div>
</div>
`;
